java-checks/src/main/java/org/sonar/java/checks/unused/UnusedPrivateFieldCheck.java

```diff
@@ -64,6 +64,13 @@ public class UnusedPrivateFieldCheck extends IssuableSubscriptionVisitor {
   private static final String DEFAULT_IGNORE_ANNOTATIONS_KEY = "ignoreAnnotations";
   private static final String DEFAULT_IGNORE_ANNOTATIONS_DESCRIPTION = "Ignore annotations with next names (fully qualified class names separated with \",\").";
 
+  private static final List<String> OWNER_CLASS_ALLOWED_ANNOTATIONS = List.of(
+    "lombok.Data",
+    "lombok.Getter",
+    "lombok.Setter",
+    "lombok.AllArgsConstructor"
+  );
```

Comment:
This should be a set, otherwise the `contains` method in line 200 executes with linear effort.
```suggestion
  private static final Set<String> OWNER_CLASS_ALLOWED_ANNOTATIONS = Set.of(
    "lombok.Data",
    "lombok.Getter",
    "lombok.Setter",
    "lombok.AllArgsConstructor"
  );
```